src/mfpbench/taskset_tabular/processing/process.py

```diff
@@ -0,0 +1,324 @@
+        )
+        if cache_path is not None:
+            cache_path.parent.mkdir(parents=True, exist_ok=True)
+            if cache_path.exists():
+                print("Loading cache from", cache_path)  # noqa: T201
+                df = pd.read_parquet(cache_path)
+                return cls(task=task, optimizer=optimizer, df=df)
+
+        # Get the curves
+        path = "/".join(["gs://gresearch/task_set_data", task, f"{name}.npz"])
+        print(f"Downloading curves for {task} {optimizer} from {path}")  # noqa: T201
+        try:
+            with GFile(path, "rb") as file:
+                cc = np.load(file)
+                steps = cc["xs"]
+                ys = cc["ys"]
+        except Exception as e:  # noqa: BLE001
+            raise ValueError(f"Failed for {task}-{optimizer} curves") from e
+
+        hps = hyperparameter_from_optimizer(optimizer, cache_dir=cache_dir)
+        hps = hps.droplevel("optimizer")  # Don't need the optimizer name in the index
+
+        curves = cls._extract_curves(ys)
+        curve_df = curves.df()
+
+        # The both join on the `"config_id"` index
+        joint_table = hps.join(curve_df)
+
+        # Right now, the "train", "valid1", "valid2", "test" columns are lists of size
+        # 51, which indicates it's loss as steps.
+        # We assign the "step" column to match this before we explode the lists, one
+        # full `"step"` list for each row.
+        joint_table["step"] = [steps] * len(joint_table)
+
+        # We also assign an "epoch" to be able to be inline with other benchmarks
+        # NOTE: This is not really an epoch and is just a proxy
+        joint_table["epoch"] = [np.arange(1, len(steps) + 1)] * len(joint_table)
+
+
+        # There's also no cost information so we just use the step as the cost
+        joint_table["train_cost"] = joint_table["step"].copy()
+        # NOTE: Use train_cost as `cost` will conflict with the Result class
+
+        # Now we explode everything out to get the following table. The HP columns
+        # will repeat mostly but that's how we need in later in the benchmark.
+        #
+        #                   | HP1 | HP2 | HP3 | ... | train | valid1 | valid2  | test | cost |  # noqa: E501
+        # config_id | epoch |
+        #     0     | 1     |
+        #           | 2     |
+        #           | ...   |
+        #           | 51    |
+        #     1     | 1     |
+        #           | 2     |
+        #           | ...   |
+        #           | 51    |
+        df = (
+            joint_table.explode(
+                [
+                    "train_loss",
+                    "valid1_loss",
+                    "valid2_loss",
+                    "test_loss",
+                    "train_cost",
+                    "step",
+                    "epoch",
+                ],
+            )
+            .reset_index()
+            .convert_dtypes()
+            .set_index(["config_id", "epoch"])
+            .sort_index()
+        )
```

Comment:
If there neeeds to be nan handling in anyway, you can do so from this point @Neeratyoy

Reply:
Added [here](https://github.com/automl/mf-prior-bench/blob/b334549904008979226d2bd7fa6a19c81dd39d59/src/mfpbench/taskset_tabular/benchmark.py#L1104).